### PR TITLE
[HatoholArmPluginGateHAPI2] Implement time-out feature for calling plugin's methods

### PR DESCRIPTION
**Pull request by ashie**
In the current implementation, HatoholArmPluginGateHAPI2 waits a response from a plugin forever
when it doesn't respond to a request.
This modification let it to give up waiting a response after 60 seconds are passed. 

issue #1458

### CODE REVIEW

File: server/common/HatoholArmPluginInterfaceHAPI2.cc [hunk 1]
```diff
@@ -370,22 +387,60 @@ struct HatoholArmPluginInterfaceHAPI2::Impl
 	void queueProcedureCallback(const string id,
 				    ProcedureCallbackPtr callback)
 	{
-		m_procedureCallbackMap[id] = callback;
+		const int PROCEDURE_TIMEOUT_MSEC = 60 * 1000;
```
Comment (cosmo0920) [inline server/common/HatoholArmPluginInterfaceHAPI2.cc]: [nitpick] This constants can make `constexpr`.

File: server/common/HatoholArmPluginInterfaceHAPI2.cc [hunk 2]
```diff
@@ -370,22 +387,60 @@ struct HatoholArmPluginInterfaceHAPI2::Impl
...
+
+		lock_guard<mutex> lock(m_procedureMapMutex);
+
+		ProcedureCallContext *context = new ProcedureCallContext();
+		context->m_impl = this;
+		context->m_callback = callback;
+		Impl *m_impl;
```
Comment (cosmo0920) [inline server/common/HatoholArmPluginInterfaceHAPI2.cc]: It seems that this variable is needless.

Comment (ashie) [inline server/common/HatoholArmPluginInterfaceHAPI2.cc]: Oops, thank you for your comment!

**Comment by kz0817** [comment]
It sounds good. Is there any reason for being left for several days ?
If no special reason, I will merge it.

**Comment by ashie** [comment]
No reason.

**Comment by ashie** [comment]
Thanks!

**Pull request merged**